### PR TITLE
Add new properties to inetutils: useOnlySiteLocalInterfaces and preferredNetworks

### PR DESCRIPTION
**Pull request by tjuchniewicz**
Fixes gh-114

### CODE REVIEW

File: spring-cloud-commons/src/main/java/org/springframework/cloud/commons/util/InetUtils.java [hunk 1]
```diff
@@ -127,6 +128,22 @@ else if (result != null) {
...
 	}
 
+	boolean ignoreAddress(InetAddress address) {
+
+		if (this.properties.isUseOnlySiteLocalInterfaces() && !address.isSiteLocalAddress()) {
+			log.trace("Ignoring address: " + address.getHostAddress());
+			return true;
+		}
+	
+		for (String regex : this.properties.getPreferredNetworks()) {
```
Comment (spencergibb) [inline spring-cloud-commons/src/main/java/org/springframework/cloud/commons/util/InetUtils.java]: I'm a bit confused here. What is `prefferedNetworks` being ignored?

**Comment by pivotal-issuemaster** [comment]
@tjuchniewicz Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-commons&pullRequestId=120)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-commons&pullRequestId=120) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/faq) for frequently asked questions.

File: spring-cloud-commons/src/main/java/org/springframework/cloud/commons/util/InetUtils.java (hunk 1 above)
Comment (tjuchniewicz) [inline spring-cloud-commons/src/main/java/org/springframework/cloud/commons/util/InetUtils.java]: Look like a bug. 

```java
if (address.getHostAddress().matches(regex) || address.getHostAddress().startsWith(regex))
```

must be replaced by:

```java
if (!address.getHostAddress().matches(regex) && !address.getHostAddress().startsWith(regex))
```

Now I use only `useOnlySiteLocalInterfaces` on production. `prefferedNetworks` is planned to be used in next months.

**Comment by pivotal-issuemaster** [comment]
@tjuchniewicz Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-commons&pullRequestId=120)!

File: spring-cloud-commons/src/main/java/org/springframework/cloud/commons/util/InetUtils.java (hunk 1 above)
Comment (tjuchniewicz) [inline spring-cloud-commons/src/main/java/org/springframework/cloud/commons/util/InetUtils.java]: Fixed

**Comment by spencergibb** [comment]
Squashed and merged via 2da3fed95dc72d9098b4b7c90203df8a5a3207fe

**Pull request closed without merging**